server/etcdserver/api/etcdhttp/health.go

```diff
@@ -178,13 +180,14 @@ func checkLeader(lg *zap.Logger, srv ServerHealth, serializable bool) Health {
 	return h
 }
 
-func checkAPI(lg *zap.Logger, srv ServerHealth, serializable bool) Health {
+func checkAPI(ctx context.Context, lg *zap.Logger, srv ServerHealth, serializable bool) Health {
 	h := Health{Health: "true"}
 	cfg := srv.Config()
-	ctx, cancel := context.WithTimeout(context.Background(), cfg.ReqTimeout())
-	_, err := srv.Range(ctx, &pb.RangeRequest{KeysOnly: true, Limit: 1, Serializable: serializable})
+	ctx = srv.AuthStore().WithRoot(ctx)
```

Comment:
There is [another place of using WithRoot](https://github.com/etcd-io/etcd/blob/185ddb2e78a6f856c6151466d57ea6306ff17c57/server/etcdserver/server.go#L874), which makes sense, because it's completely triggered by etcdserver itself. But here, the health check is triggered by users of etcd. 

Shouldn't the credential be configured on the client side? But I don't see how to do it in K8s probes.

@mitake Do you have any concern on this?

Reply:
Serializable read needs to pick a key to read, if authorization is enabled there is no easy way to know which keys users has access. Having check accept permission error seems like a big downgrade as it skips most of serializable read code. 

Having an inconsistent quality and responses of check depending on user is very weird. 

I also expect most health checkers will not get admin rights, so with auth enabled this check is pretty useless.

Reply:
I think using WithRoot is fine here because the response isn't returned to a user. If a user can supply credential (and key name), just using Range for healthcheck purpose might also be possible? So I think the endpoint shouldn't require these parameters.

Reply:
> I think using WithRoot is fine here because the response isn't returned to a user

Makes sense, although I do think the users (actually administrators of the control plane) have responsibility to get the credential configured.